sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -94,6 +96,9 @@ void AddCustomerPolicies(HttpPipelinePosition position)
 
             AddCustomerPolicies(HttpPipelinePosition.PerCall);
 
+            policies.RemoveAll(static policy => policy == null);
```

Comment:
Let's move the null check to `AddCustomerPolicies`

Reply:
Turns out we need it here because Storage is passing in a null policy to the Build method